src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -323,36 +324,59 @@ FancyZones::VirtualDesktopInitialize() noexcept
     PostMessage(m_window, WM_PRIV_VD_INIT, 0, 0);
 }
 
+bool FancyZones::MoveWindowIntoLastKnownZone(HWND window, winrt::com_ptr<IZoneWindow> zoneWindow) {
+    const auto activeZoneSet = zoneWindow->ActiveZoneSet();
+    if (activeZoneSet)
+    {
+        wil::unique_cotaskmem_string guidString;
+        if (SUCCEEDED(StringFromCLSID(activeZoneSet->Id(), &guidString)))
+        {
+            auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+            std::vector<int> zoneIndexSet = fancyZonesData.GetAppLastZoneIndexSet(window, zoneWindow->UniqueId(), guidString.get());
+            if (zoneIndexSet.size() &&
+                !fancyZonesData.IsAnotherWindowOfApplicationInstanceZoned(window, zoneWindow->UniqueId()))
+            {
+                m_windowMoveHandler.MoveWindowIntoZoneByIndexSet(window, zoneIndexSet, zoneWindow);
+                fancyZonesData.UpdateProcessIdToHandleMap(window, zoneWindow->UniqueId());
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
 // IFancyZonesCallback
 IFACEMETHODIMP_(void)
 FancyZones::WindowCreated(HWND window) noexcept
 {
-    std::shared_lock readLock(m_lock);
+    // Avoid processing splash screens and already stamped (zoned) windows.
+    if (IsSplashScreen(window) &&
+        reinterpret_cast<size_t>(::GetProp(window, MULTI_ZONE_STAMP)) != 0)
+    {
+        return;
+    }
 
-    if (m_settings->GetSettings()->appLastZone_moveWindows && IsInterestingWindow(window, m_settings->GetSettings()->excludedAppsArray))
+    std::shared_lock readLock(m_lock);
+    if (m_settings->GetSettings()->appLastZone_moveWindows &&

```

Comment:
`m_settings->GetSettings()->appLastZone_moveWindows` should be the first check entering this function, any other check should be done after.

Reply:
Done.